main/glossary/README.md

```diff
@@ -67,151 +104,379 @@ synchronously compared for structural equivalence.
+- A description of this invitation's purpose.
+
+## InvitationIssuer
+
+Since [invitations](#invitation) are special cases of [payments](#payment), invitations 
+must have a dedicated [issuer](#issuer), which is the InvitationIssuer.
+
+Zoe has a single `InvitationIssuer` for its entire lifetime. By having a reference to Zoe, 
+a user can get the `InvitationIssuer`. This lets them claim any invitation they 
+receive from someone else by calling `E(invitationIssuer).claim()` with the untrusted
+invitation as the argument. During the claiming process, the invitationIssuer validates
+the invitation. A successful claim also means that invitation is exclusively yours.
+
+**Note**: Depositing into an invitation-branded purse also claims an invitation. This is
+what the [wallet](#wallet) does.
 
 ## Issuer
-Issuers are linked to a single [mint](#mint) and vice versa, so each issuer works
+Issuers are a one-to-one relationship with both a [mint](#mint) and a [brand](#brand), so each issuer works
 with one and only one asset type, such as only working with quatloos or only working
-with moola. They can create empty [purses](#purse) and [payments](#payment) for
-their asset type, but cannot mint new [amounts](#amounts). 
+with moola. This association cannot change to another type. 
 
-Issuers can also transform
+Issuers can create empty [purses](#purse) for
+their asset type, but cannot mint new [amounts](#amounts). Issuers can also transform
 payments of their asset type (splitting, combining, burning, and exclusively claiming
 payments). An issuer from a trusted source can determine if an untrusted payment of
 its asset type is valid. 
 
-See the [ERTP Guide's Issuer section](/ertp/guide/issuers-and-mints.md)
-and the [ERTP API's Issuer section](/ertp/api/issuer.md) for more information.
+For more information, see the [ERTP Guide's Issuer section](/ertp/guide/issuers-and-mints.md)
+and the [ERTP API's Issuer section](/ertp/api/issuer.md).
+
+## Keywords
+
+Keywords are unique identifiers per contract. They tie together the [proposal](#proposal), [payments](#payment) 
+to be [escrowed](#escrow), and [payouts](#payout) to the user by serving as keys for key-value pairs in various 
+records with values of [amounts](#amounts), [issuers](#issuer), etc.
 
 ## Mint
-In ERTP, mints create digital assets and are the only objects with the authority to do so. 
-Access to a mint gives you the power to create more digital assets of its type at will. Mints
-can only create one type of asset. 
 
-Mints are [issuer's](#issuer) admin facets, and there is a one-to-one relationship between an issuer and
-its mint.
+[ERTP](#ertp) has a *mint* object, which creates digital assets. [ZCF](#zcf) provides a different interface to an ERTP mint, called a
+*ZCFMint*. It doesn't matter whether you use the ERTP mint interface (henceforth called an *ERTP mint*) or the ZCFMint interface;
+quatloos created by an ERTP mint are indistinguishable from quatloos created by a ZCFMint. 
```

Comment:
Except that any particular currency will only have one of those kinds of mints. I'm not sure this is a good way to phrase it. Maybe something more like "Assets and AssetHolders created using ZcfMints are usable in all the same ways as assets created by other ERTP Mints. They interact with Purses, Payments, Brands, and Issuers in the same ways."

Reply:
Huh. Just to be sure, is it required there can be only one mint, whether ERTP or ZCF, for a given brand? If so, I should note that in the API docs.

Reply:
> is it required there can be only one mint for a given brand? 
Absolutely. A mint can be shared, but there is only one for any given brand.

Reply:
Made the suggested change, but I'm still not totally clear on the key point. Let me ask which of the following are correct:

a) For any given brand, such as Quatloos, there can only be either a ZCFMint or an ERTP mint that creates it, not both kinds of mints.

b) If you have an ERTP mint that creates Quatloos, you can associate a ZCFMint with it (how?) such that the ZCFMint serves as an interface for creating Quatloos. However, the actual creation is done by the ERTP mint.

c) The same as b), but the ZCFMint for Quatloos is created first and then an ERTP minit is somehow associated with it to be able to create Quatloos.

Reply:
Only (a) is true.  

The zcfMint is a special case so that code inside a contract can issue a new currency, distinct from any other. Having created it that way, the Issuer and brand need to be made available to anyone that will be dealing with the associated assets. The Mint will be closely held within the contract. In the case of a normal Mint, it would be closely held outside a contract, and the same statements are true about the brand, issuer, and purses and payments.

Reply:
@Chris-Hibbert I don't think that's right. There is always an ERTP mint. It's just that in some cases, there is also a ZCFMint wrapper. So if there is a ZCFMint, there is an ERTP mint within Zoe actually minting.

a) is wrong - if there is a ZCFMint there is also an ERTP mint, just within Zoe and not available to anyone else.
b) is wrong - you cannot take an ERTP mint that already exists and create the wrapper ZCFMint after the fact.
c) is wrong - you cannot get the ERTP mint from a ZCFMint. The ERTP mint is within Zoe and not available to anyone.

The actual correct version is: If you want to be able to mint digital assets and assign them to a seat directly, within contract code, you can use Zoe to make a ZCFMint. Making a ZCFMint creates an ERTP Mint within Zoe. Calls on the ZCFMint object make calls on the ERTP Mint within Zoe, but they do some extra steps, such as escrowing the digital assets within Zoe without an offer and associating the assets with a seat. The associated ERTP Mint of the ZCFMint wrapper cannot be accessed outside of Zoe, and all minting or burning must happen by the contract code calling methods on the ZCFMint.

Reply:
I like your explanation, but I'd de-emphasize the ERTP MInt: "The ZCFMint encapsulates an ERTP Mint, and ensures that all assets created by it are assigned to a zcfSeat."  

The point that Tom is trying to clarify is that the `zcf.makeZCFMint()` produces an issuerKit that clients interact with in the same way as issuerKits from `makeIssuerKit()`. The brands, issuers, and purse and payments work the same way, and are (re-)usable by the same code.

Reply:
OK, I think I'm clear on there ultimately being an ERTP mint that mints Quatloos. It may or may not be encapsulated by a ZCFMint.  But it's still not quite clear to me if/when a mint is unique. My question now could be expressed as, let's say the asset brand is Quatloos. In a given contract instance, is it possible to mint new Quatloos using both:
 mint.mintPayment(newAmount) (the ERTP method) or:

 const mySynchronousMint = await zcf.makeZCFMint('MySyncMint', 'set');
 mySynchronousMint.mintGains({ MyKeyword: amount }, seat); (the ZCFMint method)

Or do you have to set up only one or the other in a contract instance, and once one is created, whether explicitly ERTP or ZCF-encapsulating-ERTP, only that one will ever be able to create Quatloos. Or to phrase it another way, is it the case that exactly one object can execute a command to make new Quatloos, or could both of those objects above mint Quatloos?

Reply:
If a contract ever had both at the same time, they would necessarily be distinct mints. They might even claim to have the same name, but they would create payments and purses that were incompatible with one another.

Reply:
> if/when a mint is unique

In a given contract instance, you cannot ever get the ERTP mint and the ZCFMint that encapsulates it, because when a ZCFMint is created, the ERTP mint does not escape Zoe. It's not accessible to the contract.

> Or do you have to set up only one or the other in a contract instance

There's no way to setup both the ZCFMint and the ERTP mint within it in an accessible way, because the ERTP mint that a ZCFMint wraps doesn't leave Zoe.

> once one is created, whether explicitly ERTP or ZCF-encapsulating-ERTP, only that one will ever be able to create Quatloos.

Only an ERTP mint ever actually makes Quatloos. The ZCFMint provides a method for the contract to mint. That method calls `mint.mintPayment` internally in Zoe. Another way to say this is that ZCFMints are only able to mint because they call the `mintPayment` method on the ERTP mint within Zoe that they are associated with.

> exactly one object can execute a command to make new Quatloos, or could both of those objects above mint Quatloos?

Exactly one object can execute the actual command that makes quatloos, the ERTP Mint. Zoe sets up the ZCFMint such that `myZCFMint.mintGains(...` internally calls `myERTPMint.mintPayment(...`